addons/godot-firebase/dynamiclinks/dynamiclinks.gd

```diff
@@ -0,0 +1,61 @@
+# ---------------------------------------------------- #
+#                 SCRIPT VERSION = 1.0                 #
+#                 ====================                 #
+# please, remember to increment the version to +0.1    #
+# if you are going to make changes that will committed #
+# ---------------------------------------------------- #
+
+class_name FirebaseDynamicLinks
+extends Node
+
+const _authorization_header : String = "Authorization: Bearer "
+
+var request : int = -1
+
+var _dynamic_link_request_url : String = "https://firebasedynamiclinks.googleapis.com/v1/shortLinks?key=%s"
+
+var config : Dictionary = {}
+
+var auth : Dictionary
+var _request_list_node : HTTPRequest
+
+enum REQUESTS {
+    NONE = -1,
+    GENERATE
+   }
+
+func set_config(config_json : Dictionary) -> void:
+    config = config_json
+    _dynamic_link_request_url %= config.apiKey
+    _request_list_node = HTTPRequest.new()
+    _request_list_node.connect("request_completed", self, "_on_request_completed")
+    add_child(_request_list_node)
+
+var _link_request_body : Dictionary = {
+    "longDynamicLink": "",
+    "suffix": {
+        "option": ""
+    }
+}
+
+# This function is used to generate a dynamic link using the Firebase REST API
+# It will return a JSON with the shortened link
+func generate_dynamic_link(long_link : String, is_unguessable : bool) -> void:
+    request = REQUESTS.GENERATE
+    _link_request_body.longDynamicLink = long_link
+    if is_unguessable:
+        _link_request_body.suffix.option = "UNGUESSABLE"
+    else:
+        _link_request_body.suffix.option = "SHORT"
+    _request_list_node.request(_dynamic_link_request_url, [_authorization_header + auth.idtoken], true, HTTPClient.METHOD_POST, JSON.print(_link_request_body))
```

Comment:
If you're still having authorization problems, I found out not all of the services actually support the authorization header - at least the realtime database also requires it to be sent as part of the query params in the URL.

Reply:
Fixed by doing what you had mentioned